### PR TITLE
Fbi hate crime

### PR DESCRIPTION
**Pull request by Abilityguy**
Artifacts and Scripts used for the FBI Hate Crime import.
Some of the code in this PR is borrowed from #579 and #577.

cc: @va-verma @sharadshriram @rbhoot

### CODE REVIEW

File: scripts/fbi/hate_crime/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+        incident_df,
+        groupby_cols=[],
+        agg_dict={'INCIDENT_ID': 'count'},
+        multi_index=False)
+    for idx in range(len(total_incidents)):
+        total_incidents[idx], statvars = _gen_statvar_mcf(
+            total_incidents[idx], config, population_type='CriminalIncidents')
+        statvar_list.extend(statvars)
+
+    final_df = pd.concat(total_incidents[1:])
```
Comment (rbhoot) [inline scripts/fbi/hate_crime/preprocess.py:239]: this data is not written to csv. can we write it to incidents_count.csv?

File: scripts/fbi/hate_crime/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+
+# Columns which do not contribute to a constraint property value in stat var
+NONPV_COLUMNS = [
+    'MULTIPLE_OFFENSE', 'MULTIPLE_BIAS', 'INCIDENT_ID', 'DATA_YEAR',
+    'STATE_ABBR', 'AGENCY_TYPE_NAME', 'PUB_AGENCY_NAME', 'LOCATION_NAME',
+    'BIAS_AGAINST'
+]
+
+# A dict to map bias descriptions to their category
+map_dict = {
```
Comment (ajaits) [inline scripts/fbi/hate_crime/preprocess.py:44]: Is this still needed after adding biasMotivation in the config?

Comment (Abilityguy) [reply scripts/fbi/hate_crime/preprocess.py:44]: Yes, this is needed to generate the aggregates for different Bias Categories (race, religion, gender)
Eg: biasMotivation: gender (Sum of male and female biasMotivations).

File: scripts/fbi/hate_crime/preprocess.py (hunk 1 above)
Comment (Abilityguy) [reply scripts/fbi/hate_crime/preprocess.py]: Thanks for pointing it out.
Have made the fix. Can you take a look if it's ok?

**Pull request closed without merging**